### PR TITLE
Rules Singletonification And Nodelessification

### PR DESCRIPTION
**Pull request by MartinREMY42**
Rules are now full using singleton pattern
Rules no longer have a node parameter.

### CODE REVIEW

File: src/app/jscaip/tests/Rules.spec.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ class MyAbstractState extends GameStateWithTable<number> {
 
 class AbstractNode extends MGPNode<Rules<P4Move, MyAbstractState>, P4Move, MyAbstractState> {}
 
-class AbstractRules extends Rules<P4Move, MyAbstractState> {
+class AbstractRules extends Rules<P4Move, MyAbstractState> { // TODO: je le met en singleton aussi lui ?
```
Comment (acieroid) [inline src/app/jscaip/tests/Rules.spec.ts:19]: non, osef je dirais !

Comment (acieroid) [reply src/app/jscaip/tests/Rules.spec.ts:19]: sauf si ça aide pour le greppage

File: src/app/games/six/SixMinimax.ts [hunk 2]
```diff
@@ -56,9 +56,13 @@ export class SixMinimax extends AlignementMinimax<SixMove,
         if (node.gameState.turn < 40) {
             return this.getListDrop(legalLandings);
         } else {
-            return this.getListDeplacement(node.gameState, legalLandings);
+            return this.getDeplacement(node.gameState, legalLandings);
```
Comment (acieroid) [inline src/app/games/six/SixMinimax.ts:59]: getMovement?

File: src/app/games/six/SixMinimax.ts [hunk 3]
```diff
@@ -81,10 +85,15 @@ export class SixMinimax extends AlignementMinimax<SixMove,
                     const stateAfterMove: SixState = state.movePiece(move);
                     const groupsAfterMove: MGPSet<MGPSet<Coord>> = stateAfterMove.getGroups();
                     if (SixRules.isSplit(groupsAfterMove)) {
-                        for (const group of groupsAfterMove) {
-                            const subGroup: Coord = group.getAnyElement().get();
-                            const cut: SixMove = SixMove.fromCut(start, landing, subGroup);
-                            deplacements.push(cut);
+                        const groupsOfBiggestLength: MGPSet<MGPSet<Coord>> = SixRules.getBiggerGroups(groupsAfterMove);
```
Comment (acieroid) [inline src/app/games/six/SixMinimax.ts:88]: biggestGroup? Ou mieux: largestGroup

File: src/app/games/six/SixState.ts [hunk 4]
```diff
@@ -21,7 +21,14 @@ export class SixState extends OpenHexagonalGameState<Player> {
     public override readonly height: number;
 
     public static getInitialState(): SixState {
-        const board: Table<PlayerOrNone> = [[Player.ZERO], [Player.ONE]];
+        // const board: Table<PlayerOrNone> = [[Player.ZERO], [Player.ONE]];
```
Comment (acieroid) [inline src/app/games/six/SixState.ts:24]: code comment t'es !

File: src/app/games/six/SixMinimax.ts (hunk 3 above)
Comment (MartinREMY42) [reply src/app/games/six/SixMinimax.ts:88]: oui, mais au pluriel !

**Review by acieroid — APPROVED** [review]

**Pull request merged**